### PR TITLE
Feat: Script to upload to ipfs metadata for product types and products

### PR DESCRIPTION
**Pull request by danoctavian**
## Context

Issue: https://github.com/NexusMutual/smart-contracts/issues/679

## Changes proposed in this pull request

Simply add the script handling the IPFS metadata for product types and products. The script writes out the resulting IPFS hashes to files.

## Test plan

The script was run using the existing production data and uploaded and pinned the files to the nexus node.

Manual inspection of the uploaded data was done using the gateway as such:

https://api.nexusmutual.io/ipfs/QmUhPaT7MTaqUDDD3MkZaKMJjfmrUgQYHaX7cJtyT7v6f3 (example hash)

## Checklist

- [x] Rebased the base branch
- [x] Attached corresponding Github issue
- [x] Prefixed the name with the type of change (i.e. feat, chore, test)
- [x] Performed a self-review of my own code
- [x] Followed the style guidelines of this project
- [ ] Made corresponding changes to the documentation
- [x] Didn't generate new warnings
- [x] Didn't generate failures on existing tests
- [ ] Added tests that prove my fix is effective or that my feature works

## Review

When reviewing a PR, please indicate intention in comments using the following emojis:
* :cake: = Nice to have but not essential.
* :bulb: = Suggestion or a comment based on personal opinion
* :hammer: = I believe this should be changed.
* :thinking: = I don’t understand something, do you mind giving me more context?
* :rocket: = Feedback

### CODE REVIEW

File: scripts/v2-migration/products/output/productTypeIpfsHashes.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (roxdanila) [inline scripts/v2-migration/products/output/productTypeIpfsHashes.json:1]: let's use product type Ids here as well

File: scripts/v2-migration/products/input/V2 Onchain Product Info - Products.csv [hunk 2]
```diff
@@ -0,0 +1,84 @@
+Product Id,Product Address ,Name,Product Type,Yield Token Address,Cover Assets,Initial Price Ratio,Capacity Reduction Ratio,Is Deprecated,Use Fixed Price,Allowed Pools,IPFS Hash,IPFS data
```
Comment (roxdanila) [inline scripts/v2-migration/products/input/V2 Onchain Product Info - Products.csv:1]: pls only include the info you need in this file 0 i.e. the exclusions. careful that the current spreadsheet contains multiple exclusion statements - you'll have to make each statement an item in the array

File: scripts/v2-migration/products/input/V2 Onchain Product Info - Products.csv [hunk 3]
```diff
@@ -0,0 +1,84 @@
+Product Id,Product Address ,Name,Product Type,Yield Token Address,Cover Assets,Initial Price Ratio,Capacity Reduction Ratio,Is Deprecated,Use Fixed Price,Allowed Pools,IPFS Hash,IPFS data
+,for reference only,,,(leave blank if N/A),(leave blank if both ETH/DAI apply),10% for all,0% for all,(leave blank if false),(leave blank if false),(leave blank if all),,Notes
```
Comment (roxdanila) [inline scripts/v2-migration/products/input/V2 Onchain Product Info - Products.csv:2]: rename this file to `product-ipfs-data.csv`

File: scripts/v2-migration/products/upload-product-ipfs-metdata.js [hunk 4]
```diff
@@ -0,0 +1,137 @@
+require('dotenv').config();
```
Comment (roxdanila) [inline scripts/v2-migration/products/upload-product-ipfs-metdata.js:1]: rename file to `upload-to-ipfs.js`

File: scripts/v2-migration/products/output/productIpfsHashes.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (roxdanila) [inline scripts/v2-migration/products/output/productIpfsHashes.json:1]: rename file to `product-ipfs-hashes.json`

File: scripts/v2-migration/products/upload-product-ipfs-metdata.js [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+  return productTypeHash;
+}
+
+/**
+   Expected format:
+
+ Exclusions that apply but are not limited to:
+ - Losses due to a compromised wallet;
+ - Losses due to a previously disclosed vulnerability;
+ */
```
Comment (roxdanila) [inline scripts/v2-migration/products/upload-product-ipfs-metdata.js:55]: `Exclusions that apply but are not limited to:` is unnecessary.
Each bullet point should be an item in the array

Comment (danoctavian) [reply scripts/v2-migration/products/upload-product-ipfs-metdata.js:55]: Yes that is how it's stored but the original CSV followed that format.

So I am always stripping out that prefix and throwing it away.

File: scripts/v2-migration/products/input/V2 Onchain Product Info - Products.csv (hunk 2 above)
Comment (roxdanila) [reply scripts/v2-migration/products/input/V2 Onchain Product Info - Products.csv:1]: I understand, but it's pretty much impossible to visually inspect if the input is correct.

I created this sheet, that uses the data from the main one and formatted the exclusion statements using new lines for simplicity: https://docs.google.com/spreadsheets/d/1mhPPdmVyGTZHfhnCTK9pkyuVCwIFm5JtKiTa011e9g8/edit#gid=1755020585

Comment (danoctavian) [reply scripts/v2-migration/products/input/V2 Onchain Product Info - Products.csv:1]: thank you, based on discussions in DM we'll use the whole data, i agree we should strip everything from the original exclusions column and just leave one on each line like that, the parsing is incovenient and the CSV blob is very hard to read

File: scripts/v2-migration/products/upload-product-ipfs-metdata.js (hunk 6 above)
Comment (danoctavian) [reply scripts/v2-migration/products/upload-product-ipfs-metdata.js]: eliminated now from original file each exclusion is on its own line

**Review by danoctavian — COMMENTED** [review]
LGTM

**Review by roxdanila — APPROVED** [review]

**Pull request merged**